clang/lib/Driver/Driver.cpp

```diff
@@ -4057,6 +4057,13 @@ class OffloadingActionBuilder final {
         SYCLTripleList.push_back(
             C.getDriver().MakeSYCLDeviceTriple(SYCLTargetArch));
       }
+      // Emit an error if c-compilation is forced in -fsycl mode
+      if (HasValidSYCLRuntime && Args.hasArg(options::OPT_x))
+        for (StringRef XValue : Args.getAllArgValues(options::OPT_x)) {
+          if (XValue == "c" || XValue == "c-header")
+            C.getDriver().Diag(clang::diag::err_drv_fsycl_with_c_type)
+                << "-x " << XValue;
+        }
```

Comment:
Should this be moved to `Driver::CreateOffloadingDeviceToolchains()`? It seems that doing such a diagnostic would make sense before any offloading actions' construction is attempted.

Reply:
Sure, I will try moving there and test it.